### PR TITLE
refactoring and rename package to support remote update of translations

### PR DESCRIPTION
**Pull request by jaumard**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,36 @@
-# intl_flavors
-Intl utilities to easily manage localization in Dart and Flutter
+# jimtl
+
+If you are tired on manually write ARB files or the limited Flutter localization support you are in the right place.
```
Comment (grahamsmith) [inline README.md]: If you are tired of manually writing ARB files or the limited Flutter localization support you are in the right place.

File: README.md [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
+
+With this package you'll be able to:
+
+- Generate ARB file from Dart code (done by build_runner)
```
Comment (grahamsmith) [inline README.md]: - Generate an ARB file from Dart code (done by build_runner)

File: README.md [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
   build_runner: ^2.0.3
-  intl_flavors_codegen:
+  jimtl_codegen:
 ```
 
-You then have two options, use Dart code to generate your ARB file, or manage your ARB file manually.
+## Your translation with the power of intl package
 
-We recommand you to manage your localization from Dart code, but choose the method you want and follow his setup.
+Intl is a great package to deal with translation, it let me write messages, plurals and genders in an easy and safe way.
```
Comment (grahamsmith) [inline README.md]: Intl is a great package to deal with translation, it allow you to write messages, plurals and genders in an easy and safe way.

File: README.md [hunk 4]
```diff
@@ -38,13 +41,156 @@ class Translations {
 }
 ```
 
-### Flavors
+Did you notice the `@GenerateArb()` annotation? That where the magic come from.
```
Comment (grahamsmith) [inline README.md]: Did you notice the `@GenerateArb()` annotation? That where the magic come from. 🪄

File: README.md [hunk 5]
```diff
@@ -38,13 +41,156 @@ class Translations {
...
+
+Once you run you favorite build_runner command it will generate the ARB corresponding to this class.
```
Comment (grahamsmith) [inline README.md]: Once you run your favourite `build_runner` command it will generate the ARB corresponding to this class.

File: README.md [hunk 6]
```diff
@@ -38,13 +41,156 @@ class Translations {
...
+- defaultLocale: locale your working with, default to 'en'
+- defaultFlavor: default flavor of your project, default to 'default'
+- suppressMetaData: suppress meta data when generating the ARB file, default to false
+- suppressLastModified: suppress last modified when generating the ARB file, default to false
+- includeSourceText: whether to include source_text in messages, default to false
+- dir: directory where the ARB should be generated
+
+## Flutter project
+
+If you have done Flutter project with localizations before, you'll need to declare a LocalizationDelegate in your app.
```
Comment (grahamsmith) [inline README.md]: If you have an existing Flutter project with localization support, then you'll need to declare a LocalizationDelegate in your app.

File: README.md [hunk 7]
```diff
@@ -38,13 +41,156 @@ class Translations {
...
+        setState(() {});
+      },
+      // Builder to build your custom class containing your translations
+      translationsBuilder: () => Translations(),
+    ),
+]
+```
 
-Flavor ARB doesn't have to contain all the sentences, if a sentence is not present in the flavor ARB, the default sentence from the locale will be used.
+Here is a little more details of the `TranslationsDelegate` parameters:
```
Comment (grahamsmith) [inline README.md]: Here is a little more detail of the `TranslationsDelegate` parameters:

File: README.md [hunk 8]
```diff
@@ -38,13 +41,156 @@ class Translations {
...
+- `overrideCurrentLocale`: optional locale to use for your app, it will override the system locale.
+- `dataLoader`: callback to get the ARB files for a given locale and flavor, this is call by jimtl to let you provide the ARB files when needed.
+- `updateDataLoader`: optional, to keep your translations up to date from a remote server, this is the callback you need. It should returned the ARB content for a given locale and flavor or null if no update is needed.
+- `onTranslationsUpdated`: optional, once your translations have been updated remotely by `updateDataLoader` this callback is called for your to rebuild your widget and see the changes.
+- `supportedLocales`: supported locales for your app.
+- `translationsBuilder`: builder to provide your custom class containing your sentences.
+
+### Flavors specificities
+
+Flavor support are optional in this package, but it's here in case you ever need it :)
```
Comment (grahamsmith) [inline README.md]: Flavor support is optional in this package, but it's here in case you ever need it :)

File: README.md [hunk 9]
```diff
@@ -38,13 +41,156 @@ class Translations {
...
+If you call `myClass.helloWorld` with flavor1 you'll get World Hello
+If you call `myClass.helloWorld` with flavor2 you'll get Hello World (the default one)
+
+Same goes for locale, if a sentence is not present in the current locale, the default locale sentence is used.
+
+## Dart only project
+
+If you are using Dart without flutter, you can still use this package and get all the feature he bring!
+
+Firt import the dependencies:
```
Comment (grahamsmith) [inline README.md]: First

File: README.md [hunk 10]
```diff
@@ -55,19 +201,24 @@ You have the possibility to customize `GenerateIntl` annotation with the followi
...
 - codegenMode: mode to pass to underlaying intl_generator, default to 'release'
 - useDeferredLoading: either to use deferred loading for localization files, default to true
 - arbSuppressMetaData: suppress meta data when generating the ARB file, default to false
 - arbSuppressLastModified: suppress last modified when generating the ARB file, default to false
 - arbIncludeSourceText: whether to include source_text in messages, default to false
 
-## Localization from ARB
+## Limitations
+
+As this package is based on intl, it have some of his limitations, the main one is that intl use a global internal map to find translations in the correct locale.
```
Comment (grahamsmith) [inline README.md]: As this package is based on intl, it inherits some of his limitations, the main one is that intl's use a global internal map to find translations in the correct locale.

File: README.md [hunk 11]
```diff
@@ -55,19 +201,24 @@ You have the possibility to customize `GenerateIntl` annotation with the followi
...
+
+This mean that if you have multiple ARB and custom dart classes you have to be careful that keys are unique accross all your apps. Or one will be overridden by the other.
```
Comment (grahamsmith) [inline README.md]: This means that if you have multiple ARB and custom dart classes you have to be careful that keys are unique across all your apps or one will be overridden by the other.

File: README.md [hunk 12]
```diff
@@ -55,19 +201,24 @@ You have the possibility to customize `GenerateIntl` annotation with the followi
...
+
+Another limitation is that intl have no way to tell us witch keys are conresponding to witch arguments, to fix this we're using the meta data on the source (generally en) ARB file, so be sure meta data are there or you'll have a StateError.
```
Comment (grahamsmith) [inline README.md]: Another limitation is that Intl has no way to tell us which keys are corresponding to which arguments, to fix this we're using the meta data on the source (generally en) ARB file, so be sure meta data are there or you'll have a StateError.

**Review by grahamsmith — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (jaumard) [reply README.md]: 🤦🏻 🙏🏻

**Pull request merged**